code/modules/mob/living/carbon/human/life.dm

```diff
@@ -1422,9 +1416,9 @@ var/global/list/organ_damage_overlays = list(
 					see_in_dark = 8
 					see_invisible = SEE_INVISIBLE_LEVEL_ONE
 		if(mind && mind.vampire)

```

Comment:
This breaks encapsulation, you can likely fix it by adding a simple hook (well, sort of simple):
```dm
for(var/toCall in src.callOnLife)
    if(locate(toCall) && callOnLife[toCall])
        call(locate(toCall),callOnLife[toCall])()
    else callOnLife -= toCall
```
where ```callOnLife``` is a list that you add things to like this:
```dm
callOnLife += list("\ref[foo]" = "bar")
```
```foo``` is an object that's interested in doing shit every Life() tick.
```bar``` is the shit it wants to do (the name of the proc to call, the proc has to be one of foo's)
This allows you to do vamp things in Life() AND keep vamp code contained in its own files: just define a proc with everything you need to do (all the code below) and add it to ```callOnLife``` in the code that makes someone a vampire.

Note that this actually shows some positive features of DM: the above code is horrific or impossible in most other programming languages (google "smart pointers" and "function pointers" for example).